src/Apps/Legion/Hooks/useTypes.tsx

```diff
@@ -75,23 +75,32 @@ export const useTypes = () => {
         [wizardDataState]
     );
 
+    // Single item
+    const getTypeById = (typeId: string): IViewType => {
+        return types?.find((t) => t.id === typeId);
+    };
+
     return {
         /** the current list of types in the state */
         types: types,
         /**
          * Callback to add an type to the state
          * NOTE: this is not a deep add. It will only add the root level element
          */
-        addProperty: addType,
+        addType: addType,
         /**
          * Callback to update the attributes of the type.
          * NOTE: this is not a deep update. It will only reflect changes on the root level
          */
-        updateProperty: updateType,
+        updateType: updateType,
         /**
          * Callback to delete the type from state.
          * NOTE: this is not a deep update. It will only delete the root level element
          */
-        deleteProperty: deleteType
+        deleteType: deleteType,
+        /**
+         * Callback to get a type from an id
+         */
+        getTypeById: getTypeById
```

Comment:
this is an interesting idea. 🤔 I guess it makes sense over having some utility. Good add.

Reply:
this was one of the reasons why I wanted to move to hooks, have 'utility' functions in the same place where your objects are defined seemed neat.